src/lqcut.jl

```diff
@@ -0,0 +1,133 @@
+
+# Returns
+    * `result`: NamedTuple of the fit results containing values and errors
+    * `report`: NamedTuple of the fit report which can be plotted
+"""
+function fit_lq_DEP(h::Histogram, ps::NamedTuple; uncertainty::Bool=true)
+    # create pseudo priors
+    pseudo_prior = NamedTupleDist(
+                μ = Uniform(ps.peak_pos-5*ps.peak_sigma, ps.peak_pos+5*ps.peak_sigma),
+                σ = Uniform(0.5*ps.peak_sigma, 10*ps.peak_sigma),
+                n = Uniform(0.001*ps.peak_counts, 50*ps.peak_counts),
+            )
+        
+    # transform back to frequency space
+    f_trafo = BAT.DistributionTransform(Normal, pseudo_prior)
+
+    # start values for MLE
+    v_init = mean(pseudo_prior)
+
+    # create loglikehood function
+    f_loglike = let f_fit=f_lq_DEP, h=h
+        v -> hist_loglike(x -> x in Interval(extrema(h.edges[1])...) ? f_fit(x, v) : 0, h)
+    end
+
+    # MLE
+    opt_r = optimize((-) ∘ f_loglike ∘ inverse(f_trafo), f_trafo(v_init))
+
+    # best fit results
+    v_ml = inverse(f_trafo)(Optim.minimizer(opt_r))
+
+    if uncertainty
+        f_loglike_array = let f_fit=aoe_compton_signal_peakshape, h=h
+            v -> - hist_loglike(x -> x in Interval(extrema(h.edges[1])...) ? f_fit(x, v...) : 0, h)
+        end
+
+    
+        # Calculate the Hessian matrix using ForwardDiff
+        H = ForwardDiff.hessian(f_loglike_array, tuple_to_array(v_ml))
+
+        param_covariance = nothing
+        if !all(isfinite.(H))
+            @warn "Hessian matrix is not finite"
+            param_covariance = zeros(length(v_ml), length(v_ml))
+        else
+            # Calculate the parameter covariance matrix
+            param_covariance = inv(H)
+        end
+        if ~isposdef(param_covariance)
+            param_covariance = nearestSPD(param_covariance)
+        end
+        # Extract the parameter uncertainties
+        v_ml_err = array_to_tuple(sqrt.(abs.(diag(param_covariance))), v_ml)
+
+        # get p-value 
+        pval, chi2, dof = p_value(f_lq_DEP, h, v_ml)
+        # calculate normalized residuals
+        residuals, residuals_norm, _, bin_centers = get_residuals(f_lq_DEP, h, v_ml)
+
+        @debug "Best Fit values"
+        @debug "μ: $(v_ml.μ) ± $(v_ml_err.μ)"
+        @debug "σ: $(v_ml.σ) ± $(v_ml_err.σ)"
+        @debug "n: $(v_ml.n) ± $(v_ml_err.n)"
+
+        result = merge(NamedTuple{keys(v_ml)}([measurement(v_ml[k], v_ml_err[k]) for k in keys(v_ml)]...),
+                  (gof = (pvalue = pval, chi2 = chi2, dof = dof, covmat = param_covariance, 
+                  residuals = residuals, residuals_norm = residuals_norm, bin_centers = bin_centers),))
+    else
+        @debug "Best Fit values"
+        @debug "μ: $(v_ml.μ)"
+        @debug "σ: $(v_ml.σ)"
+        @debug "n: $(v_ml.n)"
+
+        result = merge(v_ml, )
+    end
+    report = (
+            v = v_ml,
+            h = h,
+            f_fit = x -> Base.Fix2(f_lq_DEP, v_ml)(x),
+        )
+    return result, report
```

Comment:
The `result` later goes directly into the metadata, so here it is also nice to put the cut value in here. Also, where does the sideband substraction happen?